src/DX12/RHI/DX12Buffer.cpp

```diff
@@ -157,17 +150,17 @@ BindlessHandle DX12Buffer::GetOrCreateBindlessView(BufferBindingUsage usage,
             srvDesc.Format = DXGI_FORMAT_UNKNOWN;
             srvDesc.ViewDimension = D3D12_SRV_DIMENSION_BUFFER;
             srvDesc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
-            srvDesc.Buffer.FirstElement = 0;
-            srvDesc.Buffer.NumElements = desc.byteSize / *strideByteSize;
-            srvDesc.Buffer.StructureByteStride = *strideByteSize;
+            srvDesc.Buffer.FirstElement = viewDesc.offsetByteSize / viewDesc.strideByteSize;
+            srvDesc.Buffer.NumElements = viewDesc.rangeByteSize / viewDesc.strideByteSize;
+            srvDesc.Buffer.StructureByteStride = viewDesc.strideByteSize;
             srvDesc.Buffer.Flags = D3D12_BUFFER_SRV_FLAG_NONE;
             break;
         case BufferBindingUsage::ByteAddressBuffer:
             srvDesc.Format = DXGI_FORMAT_R32_TYPELESS;
             srvDesc.ViewDimension = D3D12_SRV_DIMENSION_BUFFER;
             srvDesc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
-            srvDesc.Buffer.FirstElement = 0;
-            srvDesc.Buffer.NumElements = desc.byteSize / 4; // R32 elements (4 bytes each)
+            srvDesc.Buffer.FirstElement = viewDesc.offsetByteSize / 4;
+            srvDesc.Buffer.NumElements = viewDesc.rangeByteSize / 4; // R32 elements (4 bytes each)
```

Comment:
Maybe we could add getters for these in BufferViewDesc in order to avoid having to have divide by 4 or divide by viewDesc.strideByteSize, the viewDesc probably knows the binding type, right?
If the binding type (StructuredBuffer, BAB, ...) is not known it could be passed into the getter.